README.md

```diff
@@ -73,26 +104,150 @@ The system uses dedicated queues for managing different job phases:
 
 ## 🛠️ Technical Requirements
 
-### Prerequisites
+### System Dependencies
+
+> For macOS users, use `brew install` instead of `apt install` for these dependencies.
+
+- Build essentials (`build-essential`)
+- OpenSSL (`libssl-dev`)
+- Package config (`pkg-config`)
+- Python 3.9 with development files
+- GMP library (`libgmp-dev`)
+
+### Core Dependencies
+
+- [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [Rust](https://www.rust-lang.org/tools/install)
+- [Madara Node](https://github.com/madara-alliance/madara)
+  - Required for block processing
+  - Follow setup instructions at [Madara Documentation](https://github.com/madara-alliance/madara)
+- Prover Service (ATLANTIC)
+- MongoDB for job management
+- AWS services (or Localstack for local development):
+  - SQS for queues
+  - S3 for data storage
+  - SNS for alerts
+  - EventBridge for scheduling
+
+> 🚨 **Important Note**: Currently, Madara doesn't support the `get_storage_proof` endpoint.
+>
+> Therefore, you need to run Pathfinder alongside Madara:
+>
+> - Madara will run in sequencer mode
+> - Pathfinder will sync with Madara
+> - The orchestrator will use Pathfinder's RPC URL for SNOS and state update fetching
+>
+> This setup ensures proper functionality until the `get_storage_proof` endpoint is implemented in Madara.
 
-1. **Madara Node**
+## 🚀 Installation & Setup
 
-   - Required for block processing
-   - Follow setup instructions at [Madara Documentation](https://github.com/madara-alliance/madara)
+### Building from Source
 
-2. **Prover Service**
+1. **Install System Dependencies**
 
-   - ATLANTIC must be running
+   ```bash
+   # Ubuntu/Debian
+   sudo apt-get update
+   sudo apt install build-essential openssl pkg-config libssl-dev
+   sudo apt install python3.9 python3.9-venv python3.9-distutils libgmp-dev python3.9-dev
```

Comment:
maybe a comment that python3.9 is needed for building SNOS?

Reply:
It's added generally, in [here](https://github.com/madara-alliance/madara-orchestrator/tree/feat/readme?tab=readme-ov-file#system-dependencies)

Reply:
That only mentions installing python right? I was saying let's mention why 3.9 because it's an older version (because SNOS only works with that)

Reply:
done